test/e2e/common/lease.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"time"
+
+	coordinationv1 "k8s.io/api/coordination/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/utils/pointer"
+
+	"github.com/onsi/ginkgo"
+)
+
+var _ = framework.KubeDescribe("Lease", func() {
+	f := framework.NewDefaultFramework("lease-test")
+
+	// TODO(wojtek-t): Promote to Conformance once proved to be stable.
+	ginkgo.It("API should be available", func() {
+		leaseClient := f.ClientSet.CoordinationV1().Leases(f.Namespace.Name)
+
+		name := "lease"
+		lease := &coordinationv1.Lease{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: coordinationv1.LeaseSpec{
+				HolderIdentity:       pointer.StringPtr("holder"),
+				LeaseDurationSeconds: pointer.Int32Ptr(30),
+				AcquireTime:          &metav1.MicroTime{Time: time.Time{}.Add(2 * time.Second)},
+				RenewTime:            &metav1.MicroTime{Time: time.Time{}.Add(5 * time.Second)},
+				LeaseTransitions:     pointer.Int32Ptr(0),
+			},
+		}
+
+		createdLease, err := leaseClient.Create(lease)
+		framework.ExpectNoError(err, "creating Lease failed")
+
+		readLease, err := leaseClient.Get(name, metav1.GetOptions{})
+		framework.ExpectNoError(err, "couldn't read Lease")
+		framework.ExpectEqual(apiequality.Semantic.DeepEqual(lease.Spec, readLease.Spec), true)
+
+		createdLease.Spec = coordinationv1.LeaseSpec{
+			HolderIdentity:       pointer.StringPtr("holder2"),
+			LeaseDurationSeconds: pointer.Int32Ptr(30),
+			AcquireTime:          &metav1.MicroTime{Time: time.Time{}.Add(20 * time.Second)},
+			RenewTime:            &metav1.MicroTime{Time: time.Time{}.Add(50 * time.Second)},
+			LeaseTransitions:     pointer.Int32Ptr(1),
+		}
+
+		_, err = leaseClient.Update(createdLease)
+		framework.ExpectNoError(err, "updating Lease failed")
+
+		readLease, err = leaseClient.Get(name, metav1.GetOptions{})
+		framework.ExpectNoError(err, "couldn't read Lease")
+		framework.ExpectEqual(apiequality.Semantic.DeepEqual(createdLease.Spec, readLease.Spec), true)
+
```

Comment:
also exercise patch (and ensure it was effective), list, and deletecollection (can create a second one with a specific label, then filter the list/deletecollection calls to that label to scope impact)

Reply:
My feeling is that given that we don't have any logic different for this resource than for other resources we don't need to test everything.
But sure - I can add - will do it now.